lib/src/vector_math/matrix4.dart

```diff
@@ -660,27 +668,81 @@ class Matrix4 {
   /// Returns new matrix after component wise this - [arg]
   Matrix4 operator -(Matrix4 arg) => clone()..sub(arg);
 
-  /// Translate this matrix by a [Vector3], [Vector4], or x,y,z
+  /// Translate this matrix by a [Vector3], [Vector4], or x,y,z as [double]s.
+  ///
+  /// If you know the argument types in a call site, prefer [translateByDouble],
+  /// [translateByVector3], or [translateByVector4] for performance.
+  @pragma('wasm:prefer-inline')
+  @pragma('vm:prefer-inline')
   void translate(dynamic x, [double y = 0.0, double z = 0.0]) {
-    double tx;
-    double ty;
-    double tz;
-    final tw = x is Vector4 ? x.w : 1.0;
     if (x is Vector3) {
-      tx = x.x;
-      ty = x.y;
-      tz = x.z;
+      return translateByVector3(x);
     } else if (x is Vector4) {
-      tx = x.x;
-      ty = x.y;
-      tz = x.z;
+      return translateByVector4(x);
     } else if (x is double) {
-      tx = x;
-      ty = y;
-      tz = z;
+      return translateByDouble(x, y, z);
     } else {
       throw UnimplementedError();
     }
+  }
+
+  /// Translate this matrix by x, y, z.
+  void translateByDouble(double tx, double ty, double tz) {
+    final t1 = _m4storage[0] * tx +
+        _m4storage[4] * ty +
+        _m4storage[8] * tz +
+        _m4storage[12];
+    final t2 = _m4storage[1] * tx +
+        _m4storage[5] * ty +
+        _m4storage[9] * tz +
+        _m4storage[13];
+    final t3 = _m4storage[2] * tx +
+        _m4storage[6] * ty +
+        _m4storage[10] * tz +
+        _m4storage[14];
+    final t4 = _m4storage[3] * tx +
+        _m4storage[7] * ty +
+        _m4storage[11] * tz +
+        _m4storage[15];
+    _m4storage[12] = t1;
+    _m4storage[13] = t2;
+    _m4storage[14] = t3;
+    _m4storage[15] = t4;
+  }
+
+  /// Translate this matrix by a [Vector3].
+  void translateByVector3(Vector3 v3) {
+    final tx = v3.x;
```

Comment:
Can we simply:
```dart
@pragma('vm:prefer-inline')
@pragma('dart2wasm:prefer-inline')
@pragma('dart2js:prefer-inline')
void translateByVector3(Vector3 v3)
    => translateByDouble(v3.x, v3.y, v3.z);
```
?

Also in other cases: Can we structure the code in such a way that we have the implementation only once and inline in such a way that we get the equivalent as in this CL?